iopipe/tests/test_agent.py

```diff
@@ -1,104 +1,134 @@
-def handlerThatErrors(event, context):
-    raise ValueError("Behold, a value error")
+@pytest.fixture
+def handler_with_events(iopipe):
+    @iopipe.decorator
+    def _handler_with_events(event, context):
+        iopipe.log('somekey', 2)
+        iopipe.log('anotherkey', 'qualitative value')
+    return iopipe, _handler_with_events
 
 
-def setup_function():
-    handler(None, context)
+@pytest.fixture
+def handler_that_errors(iopipe):
+    @iopipe.decorator
+    def _handler_that_errors(event, context):
+        raise ValueError("Behold, a value error")
+    return iopipe, _handler_that_errors
 
 
 # N.B. this must be the first test!
-def test_coldstarts():
+def test_coldstarts(handler, mock_context):
+    iopipe, handler = handler
+    handler(None, mock_context)
     assert iopipe.report.coldstart
-    handler(None, context)
+
+    handler(None, mock_context)
     assert not iopipe.report.coldstart
 
 
-def test_client_id_is_configured():
+def test_client_id_is_configured(handler, mock_context):
+    iopipe, handler = handler
+    handler(None, mock_context)
     assert iopipe.report.client_id == 'test-suite'
 
 
-def test_function_name_from_context():
+def test_function_name_from_context(handler, mock_context):
+    iopipe, handler = handler
+    handler(None, mock_context)
     assert iopipe.report.aws['functionName'] == 'handler'
 
 
-def test_custom_metrics():
-    handlerWithEvents(None, context)
+def test_custom_metrics(handler_with_events, mock_context):
+    iopipe, handler = handler_with_events
+    handler(None, mock_context)
     assert len(iopipe.report.custom_metrics) == 2
 
 
-def test_erroring():
+def test_erroring(handler_that_errors, mock_context):
+    iopipe, handler = handler_that_errors
     try:
-        handlerThatErrors(None, context)
+        handler(None, mock_context)
     except:
         pass
     assert iopipe.report.errors['name'] == 'ValueError'
     assert iopipe.report.errors['message'] == 'Behold, a value error'
 
 
 # Advanced reporting
-def advancedHandler(event, context):
-    # make reference for testing
-    global advancedUsage
-    iopipe = IOpipe('test-suite')
-    advancedUsage = iopipe
-    timestamp = time.time()
-    report = iopipe.create_report(timestamp, context)
-    try:
-        pass
-    except Exception as e:
-        iopipe.err(e)
-    report.send()
-
-
-def test_advanced_reporting():
-    new_context = MockContext('advancedHandler', '1')
-    advancedHandler(None, new_context)
-    assert(advancedUsage.report.aws['functionName']) == 'advancedHandler'
-
-
-def advancedHandlerWithErr(event, context):
-    global advancedUsageErr
-    iopipe = IOpipe('test-suite2')
-    advancedUsageErr = iopipe
-    timestamp = time.time()
-    report = iopipe.create_report(timestamp, context)
-    iopipe.log('name', 'foo')
-    try:
-        raise TypeError('Type error raised!')
-    except Exception as e:
-        iopipe.err(e)
-    report.send()
-
-
-def test_advanced_erroring():
+@pytest.fixture
```

Comment:
These are the docs on it: https://github.com/iopipe/iopipe-python#advanced-usage, it's an API we had/have at some point that we're still supporting. I'm guessing if someone wants to control when IOpipe is instantiated/control sending, this is when they'd use this